gromacs/fileformats/blocks.py

```diff
@@ -0,0 +1,461 @@
+
+        self.cmaps     = []
+        self.pairs     = []
+        self.exclusion_numb = None  # 0, 1, 2, ..
+        self.virtual_sites3 = []
+        self.exclusions = []
+        self.settles    = []
+        self.constraints= []
+
+        self.information = {}  # like 'atoms': self.atoms
+
+        self.name = None
+
+        self._anumb_to_atom = {}
+
+
+    def anumb_to_atom(self, anumb):
+        '''Returns the atom object corresponding to an atom number'''
+
+        assert isinstance(anumb, int), "anumb must be integer"
+
+        if len(self._anumb_to_atom) == 0:   # empty dictionary
+
+            if len(self.atoms) != 0:
+                for atom in self.atoms:
+                    self._anumb_to_atom[atom.number] = atom
+                return self._anumb_to_atom[anumb]
+            else:
+                print("no atoms in the molecule")
+                return False
+
+        else:
+            if anumb in self._anumb_to_atom:
+                return self._anumb_to_atom[anumb]
+            else:
+                print("no such atom number (%d) in the molecule" % (anumb))
+                return False
+
+
+    def renumber_atoms(self):
+
+        if len(self.atoms) != 0:
+
+            # reset the mapping
+            self._anumb_to_atom = {}
+
+            for i,atom in enumerate(self.atoms):
+                atom.number = i+1   # starting from 1
+
+        else:
+            print("the number of atoms is zero - no renumbering")
+
+
+class Chain(object):
+    """
+        name    = str,
+        residues= list,
+        molecule= Molecule
+    """
+
+    def __init__(self):
+        self.residues = []
+
+
+class Residue(object):
+    """
+        name    = str,
+        number  = int,
+        chain   = Chain,
+        chain_name = str,
+        atoms   = list,
+    """
+
+    def __init__(self):
+        self.atoms  = []
+
+
+class Atom(object):
+    """
+        name    = str,
+        number  = int,
+        flag    = str,        # HETATM
+        coords  = list,
+        residue = Residue,
+        occup   = float,
+        bfactor = float,
+        altlocs = list,
+        atomtype= str,
+        radius  = float,
+        charge  = float,
+        mass    = float,
+        chain   = str,
+        resname = str,
+        resnumb = int,
+        altloc  = str,         # per atoms
+
+    """
+
+    def __init__(self):
+
+        self.coords = []        # a list of coordinates (x,y,z) of models
+        self.altlocs= []        # a list of (altloc_name, (x,y,z), occup, bfactor)
+
+
+
+
+    def get_atomtype(self):
+        if hasattr(self, 'atomtype'):
+            return self.atomtype
+        else:
+            print("atom %s doesn't have atomtype" % self)
+            return False
+
+
+class Param(object):
+    def __init__(self):
+        self.comment = None
+        self.line  = None
+        self.disabled = False
+
+    def convert(self, reqformat):
+        assert reqformat in ('charmm', 'gromacs')
```

Comment:
1. Docstring
2. This ```if``` stack looks like it will be difficult to unit test. Could we split this into several different private methods, maybe get a Converter object going to couch these methods?